Cogs.Publishers/CsSchemaPublisher.cs

```diff
@@ -244,235 +259,264 @@ public void Publish(CogsModel model)
+                        ");
                 foreach (var p in prop.DataType.Properties)
                 {
-                    builder.Append(@"
-                        if (parts[i].Contains(""" + p.Name + "\"))");
                     if (!p.MaxCardinality.Equals("1"))
                     {
                         builder.Append(@"
+                        if (parts[i].Contains(""" + p.Name + @"""))
                         {
                             " + name + "." + p.Name + " = new List<" + p.DataTypeName + @">();
                             i++;
                             while (i < parts.Length && (string.IsNullOrWhiteSpace(parts[i].Trim().Replace(""\"""", """")) || (this.GetType().GetProperties().Where(x => parts[i].Trim().Replace(""\"""", """").
                                 ToLower().Equals(x.Name.ToLower())).ToList().Count == 0 && !""yearmonthdaydatetimeanyuricogsdate"".Contains(parts[i].Trim().Replace(""\"""", """").ToLower()))))
                             {
-                                if(!string.IsNullOrWhiteSpace(parts[i])) { obj." + p.Name + ".Add(" + ReusableTypeConvert(p.DataTypeName, true, model) + @"); }
+                                if(!string.IsNullOrWhiteSpace(parts[i])) { " + name + "." + p.Name + ".Add(" + ReusableTypeConvert(p.DataTypeName, true, model) + @"); }
                                 i++;
                             }
                         }");
                     }
                     else
                     {
-                        builder.Append(" { obj." + p.Name + " = " + ReusableTypeConvert(p.DataTypeName, false, model) + @"; }");
-
+                        builder.Append(@"
+                        if (parts[i].Contains(""" + p.Name + "\")) { " + name + "." + p.Name + " = " + ReusableTypeConvert(p.DataTypeName, false, model) + @"; }");
                     }
                 }
                 builder.Append(@"
                         i++;
                     }
-                    if (obj != null) { " + name + @".Add(obj); }
-                }
+                    reusablesInitialized++;
+                }");
             }
-            thisObj = false;");
-            }
-            else
+            return builder.ToString();
+        }
+
+
+        private void InitializeReusableList(Property prop, CogsModel model, StringBuilder main, string name, string type)
+        {
+            StringBuilder subs = new StringBuilder();
+            main.Append("$##private int Initialize" + name + "(List<" + type + "> list, string[] parts, int i)$##{$###" + type + @" obj = null;
+            i++;
+            while (i < parts.Length && (string.IsNullOrWhiteSpace(parts[i].Trim().Replace(""\"""", """")) || (this.GetType().GetProperties().Where(x => parts[i].Trim().Replace(""\"""", """").
+                ToLower().Equals(x.Name.ToLower())).ToList().Count == 0 && !""yearmonthdaydatetimeanyuricogsdate"".Contains(parts[i].Trim().Replace(""\"""", """").ToLower()))))
```

Comment:
Compare against each simple type name individually. Perhaps make a method for it?